### PR TITLE
ignore empty.lua and empty scripts

### PR DESCRIPTION
**Pull request by aronwk-aaron**
and return early if it's an ignored script
tested that ignored scripts are still ignored and not ignored scripts still load properly.

### CODE REVIEW

**Comment by aronwk-aaron** [comment]
definitely open to improvements, not sure if searching the vector of strings is the most efficient way to do that, but form cursory searches and tests say this is good enough since the the vector that is being searched is finite and not large

**Comment by EmosewaMC** [comment]
When using std::find use a list instead.

File: dScripts/CppScripts.cpp [hunk 1]
```diff
@@ -308,6 +314,9 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 
 	script = invalidToReturn;
 
+	// check for ignored scripts and return early
+	if (std::find(m_IgnoredScripts.begin(), m_IgnoredScripts.end(), scriptName) != m_IgnoredScripts.end()) return script;
```
Comment (EmosewaMC) [inline dScripts/CppScripts.cpp:318]: We should not be returning early here.  If returning early then we would not cache these scripts for faster lookup for other objects.  See line 868

**Review by EmosewaMC — CHANGES_REQUESTED** [review]

File: dScripts/CppScripts.cpp (hunk 1 above)
Comment (aronwk-aaron) [reply dScripts/CppScripts.cpp:318]: ah will update.

Comment (EmosewaMC) [reply dScripts/CppScripts.cpp:318]: IMHO it's better to just keep the current else if standard we have for script lookups than to iterate over a list, creating unneeded overhead.  At the end of the day it's still just string comparisons so adding a list of strings just seems extra.

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**